src/nitrestful.nit

```diff
@@ -0,0 +1,252 @@
+	var opt_dir: OptionString = new OptionString("Output directory", "--dir")
+
+	redef init
+	do
+		option_context.add_option(opt_output, opt_dir)
+		super
+	end
+end
+
+redef class MType
+	# Write code in `template` to parse the argument `arg_name` to this parameter type
+	private fun gen_arg_convert(template: Template, arg_name: String)
+	do
+		if self.name == "String" or self.name == "nullable String" then
+			# String are used as is
+			template.add """
+			var out_{{{arg_name}}} = in_{{{arg_name}}}
+"""
+		else
+			# Deserialize everything else
+			template.add """
+			var out_{{{arg_name}}} = deserialize_arg(in_{{{arg_name}}})
+"""
+		end
+	end
+
+	# Does this parameter type needs to be checked before calling the method?
+	#
+	# Some nullable types do not need to be check as `null` values are acceptable.
+	private fun needs_type_check: Bool do return true
+end
+
+redef class MNullableType
+	redef fun needs_type_check do return name != "nullable String" and name != "nullable Object"
+end
+
+var toolcontext = new ToolContext
+toolcontext.tooldescription = """
+Usage: nitrestful [OPTION] module.nit [other_module.nit [...]]
+Generates the boilerplate code to link RESTful request to static Nit methods."""
+
+toolcontext.process_options args
+var arguments = toolcontext.option_context.rest
+
+# Check options
+if toolcontext.opt_output.value != null and toolcontext.opt_dir.value != null then
+	print "Error: cannot use both --dir and --output"
+	exit 1
+end
+if arguments.length > 1 and toolcontext.opt_output.value != null then
+	print "Error: --output needs a single source file. Do you prefer --dir?"
+	exit 1
+end
+
+var model = new Model
+var modelbuilder = new ModelBuilder(model, toolcontext)
+
+var mmodules = modelbuilder.parse(arguments)
+modelbuilder.run_phases
+var first_mmodule = mmodules.first
+
+# Name of the support module
+var module_name
+
+# Path to the support module
+var module_path = toolcontext.opt_output.value
+if module_path == null then
+	module_name = "{first_mmodule.name}_rest"
+	module_path = "{module_name}.nit"
+
+	var dir = toolcontext.opt_dir.value
+	if dir != null then module_path = dir.join_path(module_path)
+else if module_path == "stdout" then
+	module_name = "{first_mmodule.name}_rest"
+	module_path = null
+else if module_path.has_suffix(".nit") then
+	module_name = module_path.basename(".nit")
+else
+	module_name = module_path.basename
+	module_path += ".nit"
+end
+
+var nit_module = new NitModule(module_name)
+nit_module.header = """
+# This file is generated by nitrestful
+# Do not modify, instead refine the generated services.
+"""
+
+for mmod in mmodules do
+	nit_module.imports.add mmod.name
+end
+
+var phase = toolcontext.restful_phase
+assert phase isa RestfulPhase
+
+for mclass in phase.restful_classes do
+
+	var t = new Template
+	nit_module.content.add t
+
+	t.add """
+redef class {{{mclass}}}
+	redef fun answer(request, truncated_uri)
+	do
+		var verbs = truncated_uri.split("/")
```

Comment:
Maybe rename `verbs`  into `ressources` since @jpcaissy said it was the right terminology